### PR TITLE
Updates for changes to content filtering in OpenDDS

### PR DESCRIPTION
**Pull request by mitza-oci**

### CODE REVIEW

File: src/topic_monitor.h [hunk 1]
```diff
@@ -153,6 +155,42 @@ class TopicMonitor
     /// The topic extensibility
     OpenDDS::DCPS::Extensibility m_extensibility;
 
+#if OPENDDS_MAJOR_VERSION == 3 && OPENDDS_MINOR_VERSION >= 24
+    struct FilterTypeSupport : OpenDDS::DCPS::TypeSupportImpl {
+      FilterTypeSupport(const DynamicMetaStruct& metastruct, OpenDDS::DCPS::Extensibility exten);
```
Comment (sonndinh) [inline src/topic_monitor.h:160]: We probably need a style guide for this repo for things like indentation in the future.

**Review by sonndinh — APPROVED** [review]

File: src/topic_monitor.h (hunk 1 above)
Comment (mitza-oci) [reply src/topic_monitor.h:160]: That's a good point.  At least for now I assume I can copy/paste code from OpenDDS without worrying about it.

File: src/topic_monitor.cpp [hunk 2]
```diff
@@ -298,6 +303,65 @@ void TopicMonitor::unpause()
 }
 
 
+#if OPENDDS_MAJOR_VERSION == 3 && OPENDDS_MINOR_VERSION >= 24
```
Comment (simpsont-oci) [inline src/topic_monitor.cpp:306]: Do we need these guards? We haven't officially released a version for opendds-monitor... I assume that once we're ready to do that we can just make it a requirement for it to be compiled with the minimum appropriate version of OpenDDS. Or are we trying to support any version after the release of DynamicData?

File: src/topic_monitor.h (hunk 1 above)
Comment (simpsont-oci) [reply src/topic_monitor.h:160]: I agree, we should probably agree on some guidelines (and/or a clang-format style / configuration so we can easily enforce it), but I will say that so far I have been taking some care to make sure we're following established patterns, such as 4-spaces per indentation.

Comment (mitza-oci) [reply src/topic_monitor.h:160]: I was also following established patterns, just different ones.
If we had a `.clang-format` file in the repo I'd be happy to use it.

File: src/topic_monitor.cpp (hunk 2 above)
Comment (mitza-oci) [reply src/topic_monitor.cpp:306]: Based on the CI setup, my goal was to have this monitor revision compile with both the latest release of OpenDDS and the current master branch.  It doesn't have to be that way, but it's not difficult to do and it's easy to determine when the conditionals can be dropped based on the OpenDDS release version.

File: src/topic_monitor.h (hunk 1 above)
Comment (mitza-oci) [reply src/topic_monitor.h:160]: Or we could look into https://editorconfig.org/ which seems like it has good support across vim/emacs/VS/etc.

**Pull request merged**